base_custom_attributes/custom_attributes.py

```diff
@@ -0,0 +1,449 @@
+        help="If serialized, the field will be stocked in the serialized "
+        "field: attribute_custom_tmpl or attribute_custom_variant "
+        "depending on the field based_on")
+    option_ids = fields.One2many(
+        comodel_name='attribute.option',
+        inverse_name='attribute_id',
+        string='Attribute Options')
+    create_date = fields.Datetime(
+        string='Created date',
+        readonly=True)
+    relation_model_id = fields.Many2one(
+        comodel_name='ir.model',
+        string='Model')
+    required_on_views = fields.Boolean(
+        string='Required (on views)',
+        help="If activated, the attribute will be mandatory on the views, "
+        "but not in the database")
+
+    _defaults = {
+        'model_id': _get_default_model
+    }
+
+    @api.model
+    def create(self, vals):
+        """ Create an attribute.attribute
+
+        When a `field_id` is given, the attribute will be linked to the
+        existing field. The use case is to create an attribute on a field
+        created with Python `fields`.
+
+        """
+        if vals.get('field_id'):
+            # when a 'field_id' is given, we create an attribute on an
+            # existing 'ir.model.fields'.  As this model `_inherits`
+            # 'ir.model.fields', calling `create()` with a `field_id`
+            # will call `write` in `ir.model.fields`.
+            # When the existing field is not a 'manual' field, we are
+            # not allowed to write on it. So we call `create()` without
+            # changing the fields values.
+            field_obj = self.env['ir.model.fields']
+            field = field_obj.browse(vals['field_id'])
+            if vals.get('serialized'):
+                raise orm.except_orm(
+                    _('Error'),
+                    _("Can't create a serialized attribute on "
+                      "an existing ir.model.fields (%s)") % field.name)
+            if field.state != 'manual':
+                # the ir.model.fields already exists and we want to map
+                # an attribute on it. We can't change the field so we
+                # won't add the ttype, relation and so on.
+                return super(AttributeAttribute, self).create(vals)
+
+        if vals.get('relation_model_id'):
+            relation = self.env['ir.model'].browse(
+                vals.get('relation_model_id')).model
+        else:
+            relation = 'attribute.option'
+        if vals['attribute_type'] == 'select':
+            vals['ttype'] = 'many2one'
+            vals['relation'] = relation
+        elif vals['attribute_type'] == 'multiselect':
+            vals['ttype'] = 'many2many'
+            vals['relation'] = relation
+            vals['serialized'] = True
+        else:
+            vals['ttype'] = vals['attribute_type']
+
+        if vals.get('serialized'):
+            field_obj = self.pool.get('ir.model.fields')
+            serialized_ids = field_obj.search([
```

Comment:
You should rename serialized_ids to serialized_recs as search will return you record set